zerver/tests/test_hooks.py

```diff
@@ -6,6 +6,27 @@
 import ujson
 from six.moves import urllib
 
+class WebhookTestCase(AuthedTestCase):
+    STREAM_NAME = None
+    TEST_USER_EMAIL = None
+    URL_TEMPLATE = None
+    FIXTURE_DIR_NAME = None
+
+    def setUp(self):
+        self.url = self.build_webhook_url()
+
+    def send_and_test_message(self, fixture_name, expected_subject, expected_message):
+        payload = ujson.dumps(ujson.loads(self.fixture_data(self.FIXTURE_DIR_NAME, fixture_name)))
+        msg = self.send_json_payload(self.TEST_USER_EMAIL, self.url, payload,
+                                     self.STREAM_NAME, content_type="application/json")
+
+        self.assertEqual(msg.subject, expected_subject)
+        self.assertEqual(msg.content, expected_message)
```

Comment:
I think we might have at least one integration that uses private messages for some cases (Yo, off the top of my head), so we may need an option for that.